### ISSUE TITLE
error while doing test on unet mixed-precision 

### ISSUE THREAD

**Issue by Jonxjdong** [openvinotoolkit/nncf#264]
The unet I used is the one in the examples 

configuration file is as below, 

```
{
    "model": "unet",
    "dataset" : "camvid",

    "preprocessing": {
        "resize": {
            "height": 368,
            "width": 480
        },
        "normalize":
        {
            "mean": [0.39068785, 0.40521392, 0.41434407],
            "std" : [0.29652068, 0.30514979, 0.30080369]
        }
    },


    "input_info": {
        "sample_size": [1, 3, 368, 480]
    },

    "weighing": "mfb",

    "optimizer": {
        "type": "Adam",
        "optimizer_params": {
            "lr": 5.0e-5,
            "weight_decay": 2.0e-4
        },
        "schedule_type": "step",
        "step": 100,
        "gamma": 0.1
    },

    "model_params": {
        "input_size_hw": [368, 480]
    },
    "compression": {
        "algorithm": "quantization", 
        "weights": {
            "mode": "asymmetric",
            "per_channel": true
        },
        "activations": {
            "mode": "asymmetric"
        },
        "quantize_inputs": true,
        "initializer": {
            "precision": {
                "type": "hawq",
                "bits": [4, 8],
                "num_data_points": 100,
                "iter_number": 200,
                "tolerance": 1e-4
            }
        }
        
    }, 
    "target_device": "NONE"
}
```

The training mode works fine, and I got a quantized weight with good mIoU. 
While in test mode I got the following error, 

```
Exception has occurred: ValueError
Specified non-manual precision initialization in the NNCF config, but the initializing data loader and loss criterion are not provided as an extra struct. Refer to `NNCFConfig.register_extra_structs` and the `QuantizationPrecisionInitArgs` class
```

The parameters I used in to run are as follow, 

```
python examples/semantic_segmentation/main.py \
        -m test \
        --config ./examples/semantic_segmentation/configs/unet_camvid_int8_mp.json \
        --data ./CamVid/ \
        --resume ./runs/unet_camvid_mixed_int/2020-11-09__03-28-17/unet_camvid_mixed_int_best.pth \
        --workers 0 \
```

Is there any thing incorrect in my config file or parameters?

**Comment by ljaljushkin** [comment]
Hi @Jonxjdong!

Your command line and configuration file look correct. But unfortunately, I'm not able to reproduce this error with them and with the segmentation sample from the develop branch.

I would hazard a guess you have a slightly different sample or obsolete NNCF in the python environment. Please make sure you're using the latest version.

The mentioned error message happens on precision initialization under the following conditions:
1) HAWQ-based precision initialization is enabled in the config:
```
"initializer": {
    "precision": {
        "type": "hawq"
        ...
```
2) The quantization algorithm [decided that initialization should be called](https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algo.py#L870):
```
        if is_main_process() and should_init:
            self.initialize_quantizer_params()
```
It occurs [only when resuming checkpoint is not provided](https://github.com/openvinotoolkit/nncf/blob/develop/nncf/model_creation.py#L134):
```
    should_init = resuming_state_dict is None
    compression_algo_builder_list = create_compression_algorithm_builders(config, should_init=should_init)
```
3) Criterion and data_loader are not provided: 
```
    nncf_config = register_default_init_args(nncf_config, train_loader, criterion, criterion_fn, config.device)
```

In your case, the 2nd condition is not fulfilled:  the resuming checkpoint is specified.
Even if you passed `--weights` instead of `--resume`, the 3rd condition is still not fulfilled, as [the segmentation sample registers all required structures](https://github.com/openvinotoolkit/nncf/blob/develop/examples/semantic_segmentation/main.py#L486) for initialization. So the precision initialization shouldn't happen, which is expected behavior for the test mode.

**Comment by Jonxjdong** [comment]
well, after I updated the code with the latest develop version, it worked.  I think it's more likely to be the 3rd  condition when debuging.  If the `register_default_init_args()` is called,  the error of `criterion `and `data_loader` not found will not be reported.

**Comment by ljaljushkin** [comment]
closing the issue, since it works for you now. Feel free to re-open it, if the issue is still reproduced.
